spec/decorator/event/participation_decorator_spec.rb

```diff
@@ -0,0 +1,97 @@
+# encoding: utf-8
+
+#  Copyright (c) 2019, Pfadibewegung Schweiz. This file is part of
+#  hitobito_pbs and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito_pbs.
+
+require 'spec_helper'
+
+describe Event::ParticipationDecorator, :draper_with_helpers do
+
+  let(:person) { Fabricate(:person, first_name: 'John', last_name: 'Doe', title: 'Herr',
+                           nickname: nil, town: 'Hitobitostadt', birthday: '2001-01-01') }
+  let(:kind) { Fabricate(:pbs_course_kind) }
+  let(:event) { Fabricate(:pbs_course, kind: kind, has_confirmations: true,
+                          location: 'PBSikon', groups: [ groups(:bund) ]) }
+  let(:participation_roles) { [ Event::Course::Role::Participant.new ] }
+  let(:state) { 'assigned' }
+  let(:qualified) { true }
+  let(:participation) { Fabricate(:event_participation, state: state, person: person, event: event,
+                                  roles: participation_roles, qualified: qualified) }
+  let(:decorator) { Event::ParticipationDecorator.new(participation) }
+
+  describe '#has_confirmation?' do
+
+    context 'returns true when all conditions met' do
+      it { expect(decorator.has_confirmation?).to be_truthy }
+    end
+
+    context 'returns true when participant also has helper role' do
+      let(:participation_roles) { [ Event::Course::Role::Participant.new,
+                                    Event::Course::Role::Helper.new ] }
+      it { expect(decorator.has_confirmation?).to be_truthy }
+    end
+
+    context 'returns false when event is not course' do
+      let(:event) { Fabricate(:pbs_camp) }
+      it { expect(decorator.has_confirmation?).to be_falsey }
```

Comment:
ähnlich, aber eben das negative Beispiel:

```suggestion
      it { is_expected.to_not have_confirmation }
```